### ISSUE TITLE
Removing default behaviour relying on `?url=`

### ISSUE THREAD

**Issue by greut** [UnionOfRAD/lithium#340]
We've recently hit the wall because with a [third party application](http://www.kelkoo.co.uk/co_15259-overview-of-tradedoubler.html) sending us this kind of url `?tduid=123&url=/foo/bar`. It made me thinking and I'm now seeing this `?url=` thing as harmful. `REQUEST_URI` can be enough in most cases (the [_nginx_ documentation](http://lithify.me/docs/manual/configuration/servers/nginx.wiki) already uses it but not the apache one btw).

Some exemples of _breaking_ websites:
- http://www.nateabele.com/about?url=1
- http://www.totsy.com/pages/contact?url=register
## The change:

``` diff
diff --git a/action/Request.php b/action/Request.php
index 6510ae8..95eec27 100644
--- a/action/Request.php
+++ b/action/Request.php
@@ -568,9 +568,6 @@ class Request extends \lithium\net\http\Request {
                if (isset($this->_config['url'])) {
                        return rtrim($this->_config['url'], '/');
                }
-               if (!empty($_GET['url']) ) {
-                       return rtrim($_GET['url'], '/');
-               }
                if ($uri = $this->env('REQUEST_URI')) {
                        return str_replace($this->env('base'), '/', parse_url($u
                }
```
## The backward compatibility hack

If you still need that, you can tweak your index.php Request creation. It's much difficult to do it the other way round when the `$_GET['url']` is part of `Request::_url()`.

``` php
<?php

require dirname(__DIR__) . '/config/bootstrap.php';

echo lithium\action\Dispatcher::run(new lithium\action\Request(array(
    '_config' => array(
        'url' => !empty($_GET['url']) ? $_GET['url'] : '/'
    )
)));
```

I know I'm asking for a big backward incompatible change here since I assume many websites are using the default [.htaccess](https://github.com/UnionOfRAD/framework/blob/master/app/webroot/.htaccess) or suggested [Apache configuration](http://lithify.me/docs/manual/configuration/servers/apache.wiki). Please consider it for a cleaner lithium :-)

**Comment by davidpersson** [comment]
The only way I see is removing checking for `$_GET['url']` in Request and suggest the workaround @greut detailed for those who still need it:
```
<?php
require dirname(__DIR__) . '/config/bootstrap.php';
echo lithium\action\Dispatcher::run(new lithium\action\Request(array(           
   'url' => !empty($_GET['url']) ? $_GET['url'] : '/'
)));
```

**Comment by nateabele** [comment]
Resolved in bd82155402a44dfaf47c09a10270fea3e8577015.
